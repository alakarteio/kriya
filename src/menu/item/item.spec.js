/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import mockComponent from 'misc/__mocks__/component'
import Item from './item'

jest.mock('../../button', () => mockComponent('button'))

const component = (props) => {
  return renderer.create(
    <Item {...props} />,
  )
}

const snap = (props) => {
  const c = component(props)
  const tree = c.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Item', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should have a default behaviour', () => snap({}))
    it('should take a name', () => snap({ name: 'a name' }))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
