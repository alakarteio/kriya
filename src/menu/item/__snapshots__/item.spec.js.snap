// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Item graphical (JSX) should add custom className 1`] = `
<div
  className="button"
>
  {"name":"button","props":{"style":{},"className":"item custom","large":true,"name":"MENU-ITEM-undefined"}}
</div>
`;

exports[`common/Item graphical (JSX) should add custom style 1`] = `
<div
  className="button"
>
  {"name":"button","props":{"style":{"backgroundColor":"red"},"className":"item","large":true,"name":"MENU-ITEM-undefined"}}
</div>
`;

exports[`common/Item graphical (JSX) should have a default behaviour 1`] = `
<div
  className="button"
>
  {"name":"button","props":{"style":{},"className":"item","large":true,"name":"MENU-ITEM-undefined"}}
</div>
`;

exports[`common/Item graphical (JSX) should take a name 1`] = `
<div
  className="button"
>
  {"name":"button","props":{"style":{},"className":"item","large":true,"name":"MENU-ITEM-a name"}}
</div>
`;
